src/core/qgsgnsspositioninformation.h

```diff
@@ -0,0 +1,241 @@
+
+    /**
+     * Latitude in decimal degrees, using the WGS84 datum. A positive value indicates the Northern Hemisphere, and
+     * a negative value indicates the Southern Hemisphere.
+     */
+    double latitude() const { return mLatitude; }
+    bool latitudeValid() const { return mLatitude != 0; }
+    /**
+     * Longitude in decimal degrees, using the WGS84 datum. A positive value indicates the Eastern Hemisphere, and
+     * a negative value indicates the Western Hemisphere.
+     */
+    double longitude() const { return mLongitude; }
+    bool longitudeValid() const { return mLongitude != 0; }
+
+    /**
+     * Altitude (in meters) above or below the mean sea level.
+     */
+    double elevation() const { return mElevation; }
+    bool elevationValid() const { return mElevation != 0; }
+
+    /**
+     * Ground speed, in km/h.
+     */
+    double speed() const { return mSpeed; }
+    bool speedValid() const { return mSpeed != 0; }
+
+    /**
+     * The bearing measured in degrees clockwise from true north to the direction of travel.
+     */
+    double direction() const { return mDirection; }
+    bool directionValid() const { return mDirection != std::numeric_limits< double >::quiet_NaN(); }
+
+
+    /**
+     * Contains a list of information relating to the current satellites in view.
+     */
+    QList<QgsSatelliteInfo> satellitesInView() const { return mSatellitesInView; }
+
+    /**
+     * Dilution of precision.
+     */
+    double pdop() const { return mPdop; }
+
+    /**
+     * Horizontal dilution of precision.
+     */
+    double hdop() const { return mHdop; }
+
+    /**
+     * Vertical dilution of precision.
+     */
+    double vdop() const { return mVdop; }
+
+    /**
+     * Horizontal accuracy in meters
+     */
+    double hacc() const { return mHacc; }
+    bool haccValid() const { return mHacc != std::numeric_limits< double >::quiet_NaN(); }
+
+
+    /**
+     * Vertical accuracy in meters
+     */
+    double vacc() const { return mVacc; }
+    bool vaccValid() const { return mVacc != std::numeric_limits< double >::quiet_NaN(); }
+
+    /**
+     * The date and time at which this position was reported, in UTC time.
+     */
+    QDateTime utcDateTime() const { return mUtcDateTime; }
+    /**
+     * Fix mode (where M = Manual, forced to operate in 2D or 3D or A = Automatic, 3D/2D)
+     */
+    QChar fixMode() const { return mFixMode; }
+
+    /**
+     * Contains the fix type, where 1 = no fix, 2 = 2d fix, 3 = 3d fix
+     */
+    int fixType() const { return mFixType; }
+
+    /**
+     * GPS quality indicator (0 = Invalid; 1 = Fix; 2 = Differential, 3 = Sensitive)
+     */
+    int quality() const { return mQuality; }
+
+    /**
+     * Count of satellites used in obtaining the fix.
+     */
+    int satellitesUsed() const { return mSatellitesUsed; }
+
+    /**
+     * Status (A = active or V = void)
+     */
+    QChar status() const { return mStatus; }
```

Comment:
I think we should be using an enum instead?

Reply:
I keeped the types of the values we have in `QgsGpsInformation`... There might be better ones. Should I go over them?